apps/emqx_dashboard/src/emqx_dashboard_audit.erl

```diff
@@ -18,29 +18,98 @@
 
 -include_lib("emqx/include/logger.hrl").
 %% API
--export([log/1]).
-
-log(Meta0) ->
-    #{req_start := ReqStart, req_end := ReqEnd, code := Code, method := Method} = Meta0,
-    Duration = erlang:convert_time_unit(ReqEnd - ReqStart, native, millisecond),
-    Level = level(Method, Code, Duration),
-    Username = maps:get(username, Meta0, <<"">>),
-    From = from(maps:get(auth_type, Meta0, "")),
-    Meta1 = maps:without([req_start, req_end], Meta0),
-    Meta2 = Meta1#{time => logger:timestamp(), duration_ms => Duration},
-    Meta = emqx_utils:redact(Meta2),
-    ?AUDIT(
-        Level,
-        From,
-        Meta#{username => binary_to_list(Username), node => node()}
-    ),
-    ok.
-
-from(jwt_token) -> "dashboard";
-from(_) -> "rest_api".
-
-level(get, _Code, _) -> debug;
-level(_, Code, _) when Code >= 200 andalso Code < 300 -> info;
-level(_, Code, _) when Code >= 300 andalso Code < 400 -> warning;
-level(_, Code, _) when Code >= 400 andalso Code < 500 -> error;
-level(_, _, _) -> critical.
+-export([log/2]).
+
+%% filter high frequency events
+-define(HIGH_FREQUENCY_REQUESTS, [
+    <<"/clients/:clientid/publish">>,
```

Comment:
nit: 🤔 As simple testing, the operation_id of publish API `POST /publish` is
```
{..., "operation_id":"/publish", ...}
```

Reply:
😢 sorry,     bad gpt completion